### PR TITLE
HPCC-16086 Security permissions values are hardcoded throughout source

### PR DESCRIPTION
**Pull request by RussWhitehead**
The code for Dali, workunit and plugins includes hardcoded values for
permission settings. Instead of using mysterious values like 255, -1
and 0, we should specify the named SecAccess_\* enumerations found in
seclib.hpp

Signed-off-by: Russ Whitehead william.whitehead@lexisnexis.com

### CODE REVIEW

**Comment by RussWhitehead** [comment]
@rpastrana Please review

**Comment by rpastrana** [comment]
@RussWhitehead seems reasonable. Did not verify Enum names matched previous int values.

**Comment by RussWhitehead** [comment]
I verified the enums
@richardkchapman Please merge

**Comment by rpastrana** [comment]
@RussWhitehead if memory serves me right, there's a couple of enums with very confusing, non-descriptive names. Since you're cleaning up that code you might want to refactor the enmums and make them self documenting.

**Comment by RussWhitehead** [comment]
That would be better done with a separate Jira/PR. Do you recall which ones?

File: common/workunit/workunit.cpp [hunk 1]
```diff
@@ -3950,17 +3950,17 @@ void CLocalWorkUnit::remoteCheckAccess(IUserDescriptor *user, bool writeaccess)
...
+    int perm = SecAccess_Full;
     const char *scopename = p->queryProp("@scope");
     if (scopename&&*scopename) {
         if (!user)
             user = queryUserDescriptor();
         perm = querySessionManager().getPermissionsLDAP("workunit",scopename,user,auditflags);
         if (perm<0) {
-            if (perm==-1) 
-                perm = 255;
+            if (perm == SecAccess_Unknown)
```
Comment (richardkchapman) [inline common/workunit/workunit.cpp]: In seclib.hpp, SecAccess_Unknown is set to -255

**Comment by richardkchapman** [comment]
@RussWhitehead Some concerns/confusion over the value of SecAccess_Unknown.

Why do we have SecAccess and NewSecAccess flags?

**Comment by AttilaVamos** [comment]
@RussWhitehead I think '#include "seclib.hpp" line is missing from 'dali/base/dasess.cpp'.

**Comment by RussWhitehead** [comment]
Due to a corrupted branch I am replacing this PR with https://github.com/hpcc-systems/HPCC-Platform/pull/9010

**Pull request closed without merging**